include/ncengine/physics/SupportFunctions.h

```diff
@@ -0,0 +1,67 @@
+/**
+ * @file SupportFunctions.h
+ * @copyright Jaremie Romer and McCallister Romer 2025
+ */
+#pragma once
+
+#include "DirectXMath.h"
+
+#include <memory>
+
+namespace nc
+{
+class CookedShape;
+struct Vector3;
+
+/** @brief Wrapper around a scratch buffer used by support functions. */
+class SupportBuffer
+{
+    struct Impl;
+
+    public:
+        /** @brief Get the global SupportBuffer for use with single-threaded calls to support functions. */
+        static auto GetDefault() -> SupportBuffer&;
```

Comment:
Can we make 'return value is a reference' the exception to when we use trailing return types?
I really don't like seeing auto without & here.

Reply:
I don't understand

Reply:
If you mean do `static auto& GetDefault() ...`, that isn't allowed.

Reply:
You can do `auto&` if trailing return is omitted, but we can't do that without the definition in this file.

Reply:
ah ok - must be looking at too much VB.net these days because what you're saying now makes sense. Feel free to ignore